JMGuildSaleHistoryTracker.lua

```diff
@@ -250,6 +256,61 @@ function Indexer:addExistingDataToTheIndex()
     end
 end
 
+--[[
+
+ Sale Upgrader
+
+ Will upgrade old sales to match the newest interface
+ To fix backward compatibility of the sales
+
+ ]]
+
+local SaleUpgrader = {}
+
+---
+-- Will be called after the saved variables is loaded
+-- To check and update the sale data
+--
+function SaleUpgrader:upgrade()
+    if not SavedVariables.dataVersion then
+        SavedVariables.dataVersion = 0
+    end
+
+    while SavedVariables.dataVersion < Config.dataVersion do
+        local upgradeFunction = SaleUpgrader.upgradeFunctionVersion[SavedVariables.dataVersion]
+        upgradeFunction()
```

Comment:
I thought about that, to loop through the sales at forehand and then call the upgrade function for each sale. But that would limit me in what the function can do.

So i thought it is better to just call it and allow it to do whatever it wants.
The sales data in stored in SavedVariables which is global for this scope so the upgrade function can change that whatever he wants and not just the sales in it